### PR TITLE
Option to add timestamp to bundle version

### PR DESCRIPTION
**Pull request by tipsybsd**
Option -T adds timestamp to build_id variable.

### CODE REVIEW

**Review by ppa-arenadata — APPROVED** [review]

**Comment by dgusakov** [comment]
bender build

File: adcm_client/packer/naming_rules.py [hunk 1]
```diff
@@ -86,12 +86,18 @@ def resolve_build_id(git_data, master_branches, timestamp):
...
             else:
                 build_id = '-' + git_data['branch'].replace("-", "_").replace("/", "_")
+            if timestamp_enabled:
+                build_id = build_id + '_' + timestamp
     else:
         build_id = '-1'
+        # timestamp
+        if timestamp_enabled:
+            build_id = build_id + '_' + timestamp
+
```
Comment (Maunty) [inline adcm_client/packer/naming_rules.py:96]: I don't believe that users locally reproduce pr state, even so this is special state and he must understand why it is needed, so he wont use `-T` option.

Even it this situation may occur `-T` is just ignored in this code so we may log an warning in this case.
Any way this case may be checked separately so code will become more easy to read. 

```suggestion
    else:
        build_id = '-1'
        # timestamp
    if timestamp_enabled:
        build_id = build_id + '_' + timestamp

```

**Review by Maunty — CHANGES_REQUESTED** [review]

**Comment by tipsybsd** [comment]
bender build

**Comment by tipsybsd** [comment]
bender build

File: src/adcm_client/packer/naming_rules.py [hunk 2]
```diff
@@ -123,7 +127,10 @@ def write_version(file, old_version, new_version):
     build_id = ''
     if git_data:
         check_version(version)
-        build_id = resolve_build_id(git_data, master_branches, timestamp)
+        build_id = resolve_build_id(git_data, master_branches, timestamp, timestamp_enabled)
         write_version(bundle.file, version, version + build_id)
+    else:
+        if timestamp_enabled:
+            build_id = build_id + '_' + timestamp
```
Comment (Maunty) [inline src/adcm_client/packer/naming_rules.py:134]: by default build_id is empty string, a u sure that it is expected behavior for version to be like "_12344334534"

File: src/adcm_client/packer/naming_rules.py [hunk 3]
```diff
@@ -67,7 +67,7 @@ def check_version(version):
             "-" in position {version.index("-")}').with_traceback(sys.exc_info()[2])
 
 
-def resolve_build_id(git_data, master_branches, timestamp):
+def resolve_build_id(git_data, master_branches, timestamp, timestamp_enabled):
```
Comment (Maunty) [inline src/adcm_client/packer/naming_rules.py:70]: brakes backward compatibility for this function.

File: src/adcm_client/packer/naming_rules.py [hunk 4]
```diff
@@ -88,10 +88,14 @@ def resolve_build_id(git_data, master_branches, timestamp):
                 build_id = '-' + git_data['branch'].replace("-", "_").replace("/", "_")
     else:
         build_id = '-1'
+        # timestamp
+    if timestamp_enabled:
+        build_id = build_id + '_' + timestamp
```
Comment (Maunty) [inline src/adcm_client/packer/naming_rules.py:93]: To be honest this like here seams a little bit stange, what if git data and pr in git data, then version is 1.2.3-rc12345_12345 ?

**Review by Maunty — CHANGES_REQUESTED** [review]
No version bump for adcm_client package

File: src/adcm_client/packer/naming_rules.py (hunk 2 above)
Comment (Maunty) [reply src/adcm_client/packer/naming_rules.py:134]: i dont know if there are problem with rpm version specification (we use it for adcm)
https://docs.fedoraproject.org/en-US/packaging-guidelines/Versioning/

File: src/adcm_client/packer/naming_rules.py (hunk 3 above)
Comment (amleshkov) [reply src/adcm_client/packer/naming_rules.py:70]: `def resolve_build_id(git_data, master_branches, timestamp, *args, **kwargs)` as solution

**Comment by acmnu** [comment]
Cancel in favor to #100

**Pull request closed without merging**